build/Program.fs

```diff
@@ -76,27 +78,53 @@ let buildGithubFable() =
+                    XElement.ofStringName("TargetFramework", "netcoreapp3.1"))
+            }))
+
+let createProjectFile imports (path: string) =
+    let project = generateProjectFile imports
+    project.WriteTo(path)
+
 let integration() =
-    if Shell.Exec(Tools.dotnet, "run -- --generate", path [ solutionRoot; "src" ]) <> 0 then
+    if Shell.Exec(Tools.dotnet, "run -p Snowflaqe.fsproj -- --generate", path [ solutionRoot; "src" ]) <> 0 then
         failwith "Running Fable generation failed"
     else
-        if Shell.Exec(Tools.dotnet, "build", path [ solutionRoot; "src"; "output" ]) <> 0
+        createProjectFile ("./output/Spotify.props" |> Seq.singleton) (path [ solutionRoot; "src"; "Spotify.fsproj" ])
+        if Shell.Exec(Tools.dotnet, "build Spotify.fsproj", path [ solutionRoot; "src" ]) <> 0
         then failwith "Building generated Fable project failed"
         else
-            if Shell.Exec(Tools.dotnet, "run -- --config ./snowflaqe-fsharp.json --generate", path [ solutionRoot; "src" ]) <> 0 then
+            if Shell.Exec(Tools.dotnet, "run -p Snowflaqe.fsproj -- --config ./snowflaqe-fsharp.json --generate", path [ solutionRoot; "src" ]) <> 0 then
                 failwith "Running FSharp generation failed"
             else
-            if Shell.Exec(Tools.dotnet, "build", path [ solutionRoot; "src"; "output" ]) <> 0
+            createProjectFile ("./output/Spotify.props" |> Seq.singleton) (path [ solutionRoot; "src"; "Spotify.fsproj" ])
+            if Shell.Exec(Tools.dotnet, "build Spotify.fsproj", path [ solutionRoot; "src" ]) <> 0
             then failwith "Building generated FSharp project failed"
             else
-                if Shell.Exec(Tools.dotnet, "run -- --config ./snowflaqe-shared.json --generate", path [ solutionRoot; "src" ]) <> 0 then
+                if Shell.Exec(Tools.dotnet, "run -p Snowflaqe.fsproj -- --config ./snowflaqe-shared.json --generate", path [ solutionRoot; "src" ]) <> 0 then
                     failwith "Running Shared project generation failed"
                 else
-                    let output = List.sum [
-                        Shell.Exec(Tools.dotnet, "build", path [ solutionRoot; "src"; "output"; "shared" ])
-                        Shell.Exec(Tools.dotnet, "build", path [ solutionRoot; "src"; "output"; "fable" ])
-                        Shell.Exec(Tools.dotnet, "build", path [ solutionRoot; "src"; "output"; "dotnet" ])
-                    ]
+                    createProjectFile
+                        (seq {
+                            "./output/shared/Spotify.props"
+                            "./output/fable/Spotify.props"
+                            "./output/dotnet/Spotify.props"
+                        })
+                        (path [ solutionRoot; "src"; "Spotify.fsproj" ])
+
+                    let output =
+                        Shell.Exec(Tools.dotnet, "build Spotify.fsproj", path [ solutionRoot; "src" ])
```

Comment:
Maybe you need the `-p` here instead? Also using `./src` to generate build artifacts is not prefered because this directory is tracked and one might accidentally push generated code to github. Please use a different (git ignored) directory

Reply:
> Maybe you need the `-p` here instead?

Based on this documentation, if a project or solution is explicitly indicated, then no flag is put in front of it.



> Please use a different (git ignored) directory

Ok, I'll fix it